### DISCUSSION TITLE
go install ...clairctl - Error - "go.mod file...  contains one or more replace directives. It must not contain directives that would cause it to be interpreted differently than if it were the main module."

### DISCUSSION THREAD

**Discussion by HariSekhon** [quay/clair#1601] [General]
Installing clairctl following this part of the docs:

https://quay.github.io/clair/howto/getting_started.html#submitting-a-manifest

(can't use `go get` in newer versions of Golang 1.17+ iirc, so it's `go install`)

```shell
$ GO111MODULE=on go install github.com/quay/clair/v4/cmd/clairctl@latest

go: github.com/quay/clair/v4/cmd/clairctl@latest (in github.com/quay/clair/v4@v4.4.4):
        The go.mod file for the module providing named packages contains one or
        more replace directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module.
```

```shell
$ go version
go version go1.18.4 darwin/amd64
```

This looks like something that the clair devs need to address in the go.mod for compatability with newer versions of Golang?

**Comment by BrianSipos** [comment]
I'm running into this same issue using the `go` snap on Ubuntu 20.04.

**Comment by hdonnay** [comment]
This is fixed in `main` and so `@latest` as a version specified will work on the next minor release. The specifier `@main` works currently.
